Makefile

```diff
@@ -10,31 +10,45 @@ VERSION_PATH    := provider/pkg/version.Version
 WORKING_DIR     := $(shell pwd)
 SCHEMA_PATH     := ${WORKING_DIR}/schema.json
 
+override target := "14.15.3"
+
 generate:: gen_go_sdk gen_dotnet_sdk gen_nodejs_sdk gen_python_sdk
 
 build:: build_provider build_dotnet_sdk build_nodejs_sdk build_python_sdk
 
 install:: install_provider install_dotnet_sdk install_nodejs_sdk
 
+# Ensure all dependencies are installed
+ensure::
+	yarn install
 
 # Provider
 
-build_provider::
-	cd provider/cmd/${PROVIDER}/ && \
+build_provider:: ensure
+	pushd provider/cmd/${PROVIDER}/ && \
 		yarn install && \
-		yarn run tsc --version && \
-		yarn run tsc && \
-		cp package.json yarn.lock ${PROVIDER} ${PROVIDER}.cmd PulumiPlugin.yaml ./bin/ && \
-		sed -i.bak -e "s/\$${VERSION}/$(VERSION)/g" ./bin/package.json && \
-		rm ./bin/package.json.bak
+	popd && \
+	rm -rf dist && npx --package @vercel/ncc ncc build provider/cmd/${PROVIDER}/index.ts -o dist -m && \
```

Comment:
Curious to understand if minification obfuscates error messages at all?

Reply:
Minification doesn't change strings, only minifies variable/function names without changing any of the functionality. So I would find that highly surprising.

Reply:
Does that not have any impact on stack traces that bubble up to users in error messages?

Reply:
Since I'm not clear on the kinds of stack traces we get from the provider plugin I have removed minification